macros/multiple_databases/date_from_timestamp.sql

```diff
@@ -15,6 +15,13 @@
         else
             try_convert(date, null)
     end
+{%- elif target.type == 'databricks' -%}
+    case
+        when length({{ field }}) > 0
+            then to_date({{ field }})
+        else
+            null
```

Comment:
For the other targets, we also convert the null to the corresponding data type. Especially for Databricks this should be done to prevent any occurrence of the void data type you found.